hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -189,6 +198,21 @@ public ClientType getClientType() {
         return type;
     }
 
+    @Override
+    public String getName() {
+        return clientName;
+    }
+
+    @Override
```

Comment:
Do we need this method at all since the same can be done using the below method?

And perhaps better to wrap the collection ones; e.g. when the client is created. So you don't need to do it again and it automatically protects the collection against any form of accidental modification.